### PR TITLE
Convert Output::write's type to an options arg where verbosity can be passed in as well

### PR DESCRIPTION
**Pull request by Seldaek**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #15680 |
| License | MIT |
| Doc PR |  |

### CODE REVIEW

File: src/Symfony/Component/Console/Output/Output.php [hunk 1]
```diff
@@ -121,21 +121,39 @@ public function isDebug()
...
+        $verbosity = self::VERBOSITY_NORMAL;
+        if ($options & self::VERBOSITY_QUIET) {
+            $verbosity = self::VERBOSITY_QUIET;
+        } elseif ($options & self::VERBOSITY_VERBOSE) {
+            $verbosity = self::VERBOSITY_VERBOSE;
+        } elseif ($options & self::VERBOSITY_VERY_VERBOSE) {
+            $verbosity = self::VERBOSITY_VERY_VERBOSE;
+        } elseif ($options & self::VERBOSITY_DEBUG) {
+            $verbosity = self::VERBOSITY_DEBUG;
+        }
```
Comment (WouterJ) [inline src/Symfony/Component/Console/Output/Output.php]: these if/elseif statements can be simplified significantly:

```php
$typeCheck = self::OUTPUT_NORMAL | self::OUTPUT_RAW | self::OUTPUT_PLAIN;
$type = $typeCheck & $options ?: self::OUTPUT_NORMAL;
$verbosity = ~$typeCheck & $options ?: self::VERBOSITY_NORMAL;
```

or (less dynamic):
```php
$type = 7 & $options ?: self::OUTPUT_NORMAL;
$verbosity = -8 & $options ?: self::VERBOSITY_NORMAL;
```

Comment (Seldaek) [inline src/Symfony/Component/Console/Output/Output.php]: Yup that's a good point, thanks. I implemented it with a fully explicit version for both types and verbosities though just so the code is more clear.

**Comment by WouterJ** [comment]
:+1: 

Status: Reviewed

**Comment by fabpot** [comment]
:+1:

**Comment by dunglas** [comment]
:+1:

**Comment by fabpot** [comment]
Thank you @Seldaek.

**Pull request merged**